marketplace/deployer_helm_tiller_base/bin/post-install.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eox pipefail
+
+for i in "$@"
+do
+case $i in
+  --name=*)
+    name="${i#*=}"
+    shift
+    ;;
+  --namespace=*)
+    namespace="${i#*=}"
+    shift
+    ;;
+  *)
+    >&2 echo "Unrecognized flag: $i"
+    exit 1
+    ;;
+esac
+done
+
+[[ -z "$name" ]] && >&2 echo "--name required" && exit 1
+[[ -z "$namespace" ]] && >&2 echo "--namespace required" && exit 1
+
+app_uid="$(kubectl get "applications/$name" \
+    --namespace="$namespace" \
+    --output=jsonpath='{.metadata.uid}')"
+app_api_version="$(kubectl get "applications/$name" \
+    --namespace="$namespace" \
+    --output=jsonpath='{.apiVersion}')"
+component_kinds="$(kubectl get "applications/$name" \
+    --namespace="$namespace" \
+    --output=yaml \
+  | yaml2json \
+  | jq -r '.spec.componentKinds[]
+             | [.group, .kind]
+             | @csv' \
+  | tr -d '"' \
+  | sed 's/,/ /')"
+
+manifests_directory="$(mktemp -d)"
+echo "$component_kinds" | while read group kind; do
+  kubectl get "$kind" \
+      --namespace="$namespace" \
+      --selector="heritage=Tiller,release=$name" \
+      --output=yaml \
+    | yaml2json \
+    | jq '.items[]
+            | {
+                apiVersion: .apiVersion,
+                kind: .kind,
+                metadata: .metadata
+              }
+            | del(.metadata.creationTimestamp)
+            | .metadata.ownerReferences +=
```

Comment:
An alternate approach to assigning owner references to all objects is to not do so and use a watcher container who's [terminating lifecycle hook](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/) will call helm delete.

Essentially, we create a new deployment that is owned by the application. When the application is deleted, the deployment is also GC'ed. The pod will be signaled to terminate. The container terminating hook will run helm delete on the application.

Now there's a bit of a recursion going here when the user invokes helm delete, as the post-install hook will delete the application, which results in another helm delete operation from this mechanism. We just need to ignore the helm delete failure when this hook executes.